books/kestrel/c/syntax/input-files-doc.lisp

```diff
@@ -137,8 +136,22 @@
      (xdoc::p
       "If @(':preprocess') is not @('nil'),
        and @(':preprocess-args') input is present,
-       the argument @('-E') is passed to the preprocessor,
-       followed by the arguments in the list, in that order.")
+       the behavior depends on whether
+       it evaluates to a string list or to an omap.
+       For each file, the argument @('-E') is passed to the preprocessor.
+       If @(':preprocess-args') is a string list,
+       this list of arguments is also passed to the preprocessor,
+       following the @('-E') argument.
+       If @(':preprocess-args') is an omap,
+       then we provide the list of arguments associated with the file name
+       (again, after the @('-E') argument).
+       If the file name is not in the key set,
+       only the @('-E') argument is provided.
+       If @(':preprocess-args') is @('nil'),
+       it is technically both a string list and
+       an omap from strings to string lists.
```

Comment:
```suggestion
       an omap from strings to string lists;
```
I would put a semicolon, and change the following 'The' to 'the', just so that its clear that the next sentence applies to the case of :preprocessor-args being nil, and not in general.